Swerve2022-Imported/src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -48,15 +48,14 @@ public class Swerve extends SubsystemBase
   private double                 mLimelightVisionAlignGoal;
   private double                 mVisionAlignAdjustment;
 
-  public ProfiledPIDController   snapPIDController;
-  public PIDController           visionPIDController;
+  public ProfiledPIDController   snapPIDController   = new ProfiledPIDController(Constants.SnapConstants.kP,
```

Comment:
Is initializing these variables here serve a purpose other than readability?

Reply:
Yes there is a benefit. First, the more objects you can instantiate statically when the object is created, the better. If causes the code to have all of the memory allocated needed to run--less memory is dynamically allocated when the constructor runs (only a little bit later). Second, if someone needs to access those controllers, they are already defined anywhere in any method of the Swerve class. Otherwise, the person writing the code needs to make sure they insert code that uses it after the point where it's created. This is a minor benefit, but a new person can't accidentally add an access to xxxPIDController before it's created and have a null pointer failure.

So it will work the other way, but it leaves open the potential that someone can add code that breaks it. So the benefit is for style and maintainability--not readability.

Reply:
got it, thanks!